js/screepsstats.js

```diff
@@ -54,212 +47,183 @@ ScreepsStats.prototype.addStat = function (key, value) {
-    'bucket': Game.cpu.bucket
-  }
-
-  stats['gcl'] = {
-    'level': Game.gcl.level,
-    'progress': Game.gcl.progress,
-    'progressTotal': Game.gcl.progressTotal
-  }
-
-  if(!stats['rooms']) {
-    stats['rooms'] = {
-      'subgroups': true
+  var stats = {
+    time: new Date().toISOString(),
+    tick: Game.time,
+    cpu: {
+      limit: Game.cpu.limit,
+      tickLimit: Game.cpu.tickLimit,
+      bucket: Game.cpu.bucket
+    },
+    gcl: {
+     level: Game.gcl.level,
+     progress: Game.gcl.progress,
+     progressTotal: Game.gcl.progressTotal
     }
   }
 
+  _.defaults(stats, {
+    rooms: {
+      subgroups: true
+    }
+  })
 
-  for(var roomName in Game.rooms) {
-    var room = Game.rooms[roomName]
+  _.forEach(Game.rooms,(room) => {
 
-    if(!stats[roomName]) {
-      stats['rooms'][roomName] = {}
+    if(!stats[room.name]) {
+      stats.rooms[room.name] = {}
     }
 
-    if(!!room.controller) {
-      var controller = room.controller
+    if (_.isEmpty(room.controller)) { return }
+    var controller = room.controller
 
-      // Is hostile room? Continue
-      if(!controller.my) {
-        if(!!controller.owner) { // Owner is set but is not this user.
-          if(controller.owner.username != this.username) {
-            continue
-          }
+    // Is hostile room? Continue
+    if(!controller.my) {
+      if(!!controller.owner) { // Owner is set but is not this user.
+        if(controller.owner.username != this.username) {
+          return
         }
       }
+    }
 
-      // Collect stats
-      stats['rooms'][roomName]['level'] = controller.level
-      stats['rooms'][roomName]['progress'] = controller.progress
-
-      if(!!controller.upgradeBlocked) {
-        stats['rooms'][roomName]['upgradeBlocked'] = controller.upgradeBlocked
-      }
-
-      if(!!controller.reservation) {
-        stats['rooms'][roomName]['reservation'] = controller.reservation.ticksToEnd
-      }
-
-      if(!!controller.ticksToDowngrade) {
-        stats['rooms'][roomName]['ticksToDowngrade'] = controller.ticksToDowngrade
-      }
-
-      if(controller.level > 0) {
-
-        stats['rooms'][roomName]['energyAvailable'] = room.energyAvailable
-        stats['rooms'][roomName]['energyAvailable'] = room.energyCapacityAvailable
-
-        if(room.storage) {
-          if(!stats['storage']) {
-            stats['storage'] = {
-              'subgroups': true
-            }
-          }
-
-          stats['storage'][room.storage.id] = {}
-          stats['storage'][room.storage.id].room = room.name
-          stats['storage'][room.storage.id].store = _.sum(room.storage.store)
-          stats['storage'][room.storage.id]['resources'] = {}
-          for(var resourceType in room.storage.store) {
-            stats['storage'][room.storage.id]['resources'][resourceType] = room.storage.store[resourceType]
-            stats['storage'][room.storage.id][resourceType] = room.storage.store[resourceType]
+    // Controller
```

Comment:
_.merge ignores undefined, so in combination with the _.get(controller,'reservation.ticksToEnd') we can avoid all the undefined checks